docs/data/data-grid/export/ExportServerSideData.js

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import {
+  DataGridPro,
+  gridDataRowIdsSelector,
+  Toolbar,
+  ToolbarButton,
+  useGridApiContext,
+} from '@mui/x-data-grid-pro';
+import { createFakeServer } from '@mui/x-data-grid-generator';
+import Button from '@mui/material/Button';
+import Download from '@mui/icons-material/Download';
+
+const SERVER_OPTIONS = {
+  useCursorPagination: false,
+};
+
+const { useQuery, ...data } = createFakeServer(
+  {
+    rowLength: 10000,
+  },
+  SERVER_OPTIONS,
+);
+
+function CustomToolbar() {
+  const apiRef = useGridApiContext();
+  const [exporting, setExporting] = React.useState(false);
+
+  // Due to the mockup API, it's required to get all rows from the hook.
+  // For a real-world scenario, this line is not necessary.
+  const { rows: allExportRows } = useQuery({});
+
+  async function exportAllRows(exporter) {
+    apiRef.current.setLoading(true);
+    setExporting(true);
+
+    // Save current row IDs to restore later
+    const currentRowIds = gridDataRowIdsSelector(apiRef);
+
+    try {
+      // TODO: replace this line with real API call to your database.
+      const allRows = await Promise.resolve(allExportRows);
+
+      apiRef.current.updateRows(allRows);
+
+      exporter({
+        fileName: `full-csv-export`,
+        includeHeaders: true,
+      });
+
+      const idsToDelete = allRows
+        .map((row) => row.id)
+        .filter((id) => !currentRowIds.includes(id));
+
+      // Restore to original rows
+      apiRef.current.updateRows(
+        idsToDelete.map((id) => ({ id, _action: 'delete' })),
+      );
+    } finally {
```

Comment:
shouldn't we have a `catch` here? Even if it is only throwing we should probably add this. WDYT @MBilalShafi?